File: 0071-simplify-path/0071-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        Stack<String> stack = new Stack<String>();
        for (String c : path.split("/")) {
            if (c.equals("..") && !stack.empty()) {
                    stack.pop();
            } else if (!c.equals("..") &&!c.equals(".") && !c.equals("")) {
                stack.push(c);
            }
        }
        return "/" + String.join("/", stack);
        }
    }
        

File: 0071-simplify-path/0071-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack=[]
        skip=("..",".","")
        for c in path.split("/"):
            if stack and c == "..":
                stack.pop()
            elif c not in skip:
                stack.append(c)
        return "/"+"/".join(stack)